### PR TITLE
Ignore vanished graph location during typechecking after paste

### LINKED ISSUE #1107: Copy-pasting whole file throws exception
### Steps to Reproduce

1. Copy the following code:
```haskell
import Std.Base

class Foo:
    x :: Bar

def main:
    foo1 = Foo 3
    x1 = foo1 . x
    test = "Test me!"
    None
```
2. Open new project
3. Select the whole code
4. Press `ctrl + v`

### Expected Result

It should work

### Luna Version
1.4.101.6

### PR DESCRIPTION
**Pull request by mikusp**
### Pull Request Description

Fixes #1107.

### Checklist

- [ ] The documentation has been updated if necessary.
- [ ] The code conforms to the [Luna Style Guide](https://github.com/luna/wiki/blob/master/code-style/01.general.md) if it is Haskell.
- [ ] The code has been tested where possible.

### CODE REVIEW

File: libs/luna-empire/src/Empire/Commands/Graph.hs [hunk 1]
```diff
@@ -1332,24 +1332,30 @@ rangeToMarked code range' = (start, end)
         (start, end) = Code.viewDeltasToRealBeforeMarker code span
 
 pasteText :: GraphLocation -> [Range] -> [Text] -> Empire Text
-pasteText loc@(GraphLocation file _) ranges (Text.concat -> text) = do
-    res <- withUnit (GraphLocation file (Breadcrumb [])) $ do
+pasteText loc ranges (Text.concat -> text) = do
```
Comment (ljkania) [inline libs/luna-empire/src/Empire/Commands/Graph.hs]: If you don’t use loc for anything else then getting top consider convert in argument

File: libs/luna-empire/src/Empire/Commands/Graph.hs [hunk 2]
```diff
@@ -1332,24 +1332,30 @@ rangeToMarked code range' = (start, end)
...
+                \(e::ASTParse.SomeParserException) -> do
+                    withUnit topLoc $ do
+                        Graph.userState . Graph.code .= newCode
+                        Graph.userState . Graph.clsParseError ?= toException e
+            typecheck loc `catch` \(e::BH.BreadcrumbDoesNotExistException) ->
+                -- if after reloading, our loc no longer exists, ignore error
+                pure ()
+            resendCodeWithCursor topLoc (Just cursorPos)
+            pure newCode
+        _ -> pure ""
```
Comment (ljkania) [inline libs/luna-empire/src/Empire/Commands/Graph.hs]: Use mempty

File: libs/luna-empire/src/Empire/Commands/Graph.hs [hunk 3]
```diff
@@ -1332,24 +1332,30 @@ rangeToMarked code range' = (start, end)
...
 pasteText :: GraphLocation -> [Range] -> [Text] -> Empire Text
-pasteText loc@(GraphLocation file _) ranges (Text.concat -> text) = do
-    res <- withUnit (GraphLocation file (Breadcrumb [])) $ do
+pasteText loc ranges (Text.concat -> text) = do
+    let topLoc = GraphLocation.top loc
+    res <- withUnit topLoc $ do
         runASTOp $ forM (Safe.headMay ranges) $ \range -> do
             code <- use Graph.code
             let (start, end)   = rangeToMarked code range
             let cleanText = Code.removeMarkers text
```
Comment (ljkania) [inline libs/luna-empire/src/Empire/Commands/Graph.hs]: Not needed let

File: libs/luna-empire/src/Empire/Commands/Graph.hs [hunk 4]
```diff
@@ -1332,24 +1332,30 @@ rangeToMarked code range' = (start, end)
...
 
 pasteText :: GraphLocation -> [Range] -> [Text] -> Empire Text
-pasteText loc@(GraphLocation file _) ranges (Text.concat -> text) = do
-    res <- withUnit (GraphLocation file (Breadcrumb [])) $ do
+pasteText loc ranges (Text.concat -> text) = do
+    let topLoc = GraphLocation.top loc
+    res <- withUnit topLoc $ do
         runASTOp $ forM (Safe.headMay ranges) $ \range -> do
             code <- use Graph.code
             let (start, end)   = rangeToMarked code range
```
Comment (ljkania) [inline libs/luna-empire/src/Empire/Commands/Graph.hs]: Extra spaces

File: libs/luna-empire/src/Empire/Commands/Graph.hs (hunk 1 above)
Comment (mikusp) [reply libs/luna-empire/src/Empire/Commands/Graph.hs]: I do for `typecheck`

**Pull request merged**